Data/ByteString/Builder/RealFloat/Internal.hs

```diff
@@ -734,36 +708,22 @@ ascii_e = ord 'e'
 toAscii :: Word# -> Word#
 toAscii a = a `plusWord#` asciiRaw asciiZero
 
-data Addr = Addr Addr#
-
 -- | Index into the 64-bit word lookup table provided
 {-# INLINE getWord64At #-}
-getWord64At :: Addr# -> Int -> Word64
-getWord64At arr (I# i) =
-#if defined(WORDS_BIGENDIAN)
-   W64# (byteSwap64# (indexWord64OffAddr# arr i))
-#else
-   W64# (indexWord64OffAddr# arr i)
-#endif
+getWord64At :: Ptr Word64 -> Int -> Word64
+getWord64At (Ptr arr) (I# i) = W64# (indexWord64OffAddr# arr i)
```

Comment:
Why is it sufficient now, without `byteSwap64#`?

Reply:
Nevermind, it's because lookup tables now automatically have correct endianness.